chapters/classes.tex

```diff
@@ -351,62 +373,98 @@ \subsection{Acyclic Bindings of Constants and Parameters}\label{acyclic-bindings
 \end{lstlisting}
 \end{example}
 
-\subsection{Component Variability Prefixes}\label{component-variability-prefixes-discrete-parameter-constant}
 
-The prefixes \lstinline!discrete!, \lstinline!parameter!, \lstinline!constant! of a component declaration are called \firstuse[variability!prefix]{variability prefixes}\index{component variability}\index{declared variability}\index{variability!declared|see{declared variability}} and define in which situation the variable values of a component are initialized (see \cref{events-and-synchronization} and \cref{initialization-initial-equation-and-initial-algorithm}) and when they are changed during simulation:
+\subsection{Conditional Component Declaration}\label{conditional-component-declaration}
+
+A component declaration can have a \lstinline!condition-attribute!: \lstinline!if!~\emph{expression}.
+
+\begin{example}
+\begin{lstlisting}[language=modelica]
+  parameter Integer level(min=1)=1;
+  Motor motor;
+  Level1 component1(J=J) if level==1 "Conditional component";
+  Level2 component2 if level==2 "Conditional component";
+  Level3 component3(J=component1.J) if level<2 "Conditional component";
+  // Illegal modifier on component3 since component1.J is conditional
+  // Even if we can see that component1 always exist if component3 exist
+equation
+  connect(component1$\ldots$, $\ldots$) "Connection to conditional component 1";
+  connect(component2.n, motor.n) "Connection to conditional component 2";
+  connect(component3.n, motor.n) "Connection to conditional component 3";
+  component1.u=0; // Illegal
+\end{lstlisting}
+\end{example}
+
+The \emph{expression} must be a \lstinline!Boolean! scalar expression, and must be a parameter expression.
```

Comment:
Shouldn't this be:
```suggestion
The \emph{expression} must be a \lstinline!Boolean! scalar expression, and must be an evaluable expression.
```
?

Reply:
No, but just because we decided to only apply the new terminology to array dimensions in this PR.  As soon as this PR is merged, we should set up a separate PR for this change.

Reply:
Ok

Reply:
Good.  Resolving.